app/services/file_manager.rb

```diff
@@ -82,7 +82,30 @@ def expires_at
                 :allowed_types, :days_to_live, :encrypted_user_id_and_token
 
   def uploader
-    Storage::S3::Uploader.new(path: path_to_file, key: key, bucket: bucket)
+    Storage::S3::Uploader.new(key: key, bucket: bucket)
+  end
+
+  def encrypted_file
+    decoded_body = Base64.strict_decode64(encoded_file)
+
+    encrypted_body = Cryptography.new(
+      encryption_key: encryption_key,
+      encryption_iv: encryption_iv
+    ).encrypt(file: decoded_body)
+
+    file = Tempfile.new.binmode
+
+    file.write(encrypted_body)
+    file.rewind
+    file
+  end
```

Comment:
This method is rather long. A bit of explanation:
- `binmode` is recommended when your files aren't plain text. It reduces risk of dataloss.
- `rewind` is required so that the next time `.read` is called it returns the contents rather than an empty screen.

Using `Tempfile` here means we don't need to save to disk. There is a lot of that going on.

Reply:
It would also be nice to do the Base64 decoding in the controller, but leaving that for a separate PR.

Reply:
Removed this method